### ISSUE TITLE
[Snowflake] SYSTEM$-prefixed functions cause PRS error

### ISSUE THREAD

**Issue by miguelmmu** [sqlfluff/sqlfluff#8038]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting a Snowflake query that uses a SYSTEM$ built-in function (e.g. SYSTEM$CURRENT_USER_TASK_NAME()) inside a SELECT clause, SQLFluff fails to parse the statement. The parser probably treats the $ in the function name as a token boundary, so it cannot recognise SYSTEM$CURRENT_USER_TASK_NAME() as a single function call. This produces a PRS error.

### Expected Behaviour

SQLFluff should parse SYSTEM$-prefixed functions (such as SYSTEM$CURRENT_USER_TASK_NAME, SYSTEM$CLUSTERING_DEPTH, SYSTEM$STREAM_HAS_DATA, etc.) as valid Snowflake built-in functions. The query should parse cleanly, the explicit AS aliases should be recognised, and no PRS or other rule violations should be raised.

### Observed Behaviour

Linting the query below raises a PRS error:

```sql
SELECT
    A.COL_1,
    SYSTEM$CURRENT_USER_TASK_NAME() AS TASK_NAME,
    IFF(SYSTEM$CURRENT_USER_TASK_NAME() <> '', SYSTEM$CURRENT_USER_TASK_NAME(), CURRENT_USER()) AS LOAD_USER
FROM (SELECT ROW_NUMBER() OVER (ORDER BY SEQ4()) AS COL_1 FROM TABLE(GENERATOR(ROWCOUNT => 5))) A;
``` 
```
violations:
  - start_line_no: 3
    start_line_pos: 5
    code: AL03
    description: Column expression without alias. Use explicit `AS` clause.
    name: aliasing.expression
    warning: false
  - start_line_no: 3
    start_line_pos: 37
    code: PRS
    description: "Line 3, Position 37: Found unparsable section: 'AS TASK_NAME,\\n    IFF(SYSTEM$CURRENT_USE...'"
    name: ''
    warning: false
``` 
This query is valid Snowflake SQL and executes without any error. It returns the expected result set (TASK_NAME is empty this time because the statement is run directly by a user rather than from a task).
| COL_1 | TASK_NAME | LOAD_USER |
|--------|--------|--------|
| 1 |  | DUMMYUSER |
| 2 |  | DUMMYUSER |
| 3 |  | DUMMYUSER |
| 4 |  | DUMMYUSER |
| 5 |  | DUMMYUSER | 

### How to reproduce

Run sqlfluff lint on the SQL query above.

### Dialect

snowflake

### Version

4.2.2

### Configuration

dialect = snowflake
templater = raw
encoding = utf-8
max_line_length = 0
large_file_skip_byte_limit = 0

rules = AL02, AL03, AL04, AL05, AL08, AM01, AM02, AM04, AM05, AM06, AM07, CV05, CV07, CV08, CV12, LT07, LT10, LT11, LT13, LT14, RF01, RF02, RF04, RF05, RF06, ST02, ST03, ST04, ST07, ST08, ST10, ST11, ST12

indent_unit = space
tab_space_size = 4
indented_joins = False
indented_ctes = False
indented_using_on = True
indented_on_contents = False
indented_then = False
indented_then_contents = False
implicit_indents = allow
template_blocks_indent = True

[sqlfluff:rules:references.qualification]
subqueries_ignore_external_references = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
